pod.go

```diff
@@ -339,8 +339,6 @@ type Pod struct {
 	runPath    string
 	configPath string
 
-	url string
-
```

Comment:
The only case would be the proxy opening different sockets for different processes. Every time we call Register or Attach to reach the proxy, we receive a URL, we can assume it could be different for some reasons (maybe performance).
The main reason for this change is that in case of starting a container, we need to get the URL to start the shim. But even if I return the pod from StartContainer() API, pod.URL() returns an empty string. Indeed, the URL has been obtained when we did createContainer() from hyperstart.go, meaning there is no way we can get this URL at the time of StartContainer() in case of virtc. The only way would be to store the URL in the PodConfig, but this is not a config coming from the user, and if we set this field as private, the JSON marshalling won't work.
That's why I have thought that using the Process structure would be a good idea because we store it already, and we are covering the possibility to get different URLs from the proxy.